### PR TITLE
APS/ MMG NRE fixes

### PR DESCRIPTION
**Pull request by SuicidalInsanity**
-Add veeType = Sub for AI GUI 
-have torpedoes head to last known contact point and start circling it, instead of current sink to bottom behavior.
-Fix BurstFirelength and FireAngleoverride toggle non-function in flight
-Fix APS getting mixed in with global weapon shortnaming and overriding barrage settings
-MML toggle deploy anim toggle now affects symmetry twins
-Fix NREs with MMGs
-Fix multiple anti-ballistic APS getting assigned to the same shell if APS count > tgtShell count

### CODE REVIEW

File: BDArmory/Weapons/Missiles/BDModularGuidance.cs
Comment (BrettRyland) [inline BDArmory/Weapons/Missiles/BDModularGuidance.cs:1]: Would it be worth checking for no ignited engines in `BDModularGuidance.MissileIgnition()` after the `ExecuteNextStage()` call on line 421 and then igniting any engines (i.e., `SpawnUtils.CountActiveEngines` and `SpawnUtils.ActivateAllEngines`)?
Forgetting to activate the engine with action groups is probably something that happens fairly frequently.
Are there any situations where you wouldn't want to ignite the engine if none are activated by AG1?

File: BDArmory/UI/BDAEditorArmorWindow.cs [hunk 1]
```diff
@@ -1039,7 +1039,7 @@ private void CalculateArmorStats()
                 */
                 //armorValue = ProjectileUtils.CalculatePenetration(30, newCaliber, 0.388f, 1109, ArmorDuctility, ArmorDensity, ArmorStrength, 30, 0.8f, false);
                 armorValue = ProjectileUtils.CalculatePenetration(30, 1109, 0.388f, 0.8f, ArmorStrength, ArmorVfactor, ArmorMu1, ArmorMu2, ArmorMu3); //why is this hardcoded? it needs to be the selected armor mat's vars
-                relValue = Mathf.Round(armorValue / steelValue * 10) / 10;
+                relValue = Mathf.Round(steelValue / armorValue * 10) / 10;
```
Comment (BrettRyland) [inline BDArmory/UI/BDAEditorArmorWindow.cs:1042]: `BDAMath.RoundToUnit(steelValue / armorValue, 0.1f)` is cleaner.

File: BDArmory/Weapons/ModuleWeapon.cs [hunk 2]
```diff
@@ -1658,60 +1659,38 @@ public void PAWRefresh()
...
-                            if (weapon.Current.FireAngleOverride == false)
-                            {
-                                weapon.Current.Events["ToggleOverrideAngle"].guiName = StringUtils.Localize("#LOC_BDArmory_FireAngleOverride_Enable");// Enable Firing Angle Override
-                            }
-                            else
-                            {
-                                weapon.Current.Events["ToggleOverrideAngle"].guiName = StringUtils.Localize("#LOC_BDArmory_FireAngleOverride_Disable");// Disable Firing Angle Override
-                            }
+            FireAngleOverride = !FireAngleOverride;
+            if (FireAngleOverride == false)
```
Comment (BrettRyland) [inline BDArmory/Weapons/ModuleWeapon.cs:1663]: `if (!FireAngleOverride)` would be cleaner (like with `BurstOverride` below), or `if (fireAngleOverride)` and switch the blocks around.

**Comment by BrettRyland** [comment]
I got another exception
```csharp
[EXC 12:55:00.242] NullReferenceException: Object reference not set to an instance of an object
  BDArmory.Control.MissileFire+<GuardMissileRoutine>d__296.MoveNext () (at /home/brett/github/KSP/BDArmory/BDArmory/Control/MissileFire.cs:2354)
```
which I think is due to `foundCam` being `null`.

File: BDArmory/Weapons/Missiles/BDModularGuidance.cs
Comment (SuicidalInsanity) [reply BDArmory/Weapons/Missiles/BDModularGuidance.cs:1]: Any multistage missile that has second/etc stage engines would be a case where you wouldn't want global autoignition.
But we could add a 
```
if (StagesNumber == 1) {
    if (SpawnUtils.CountActiveEngines(vessel) < 1)
    SpawnUtils.ActivateAllEngines(vessel); }
```
condition for single stage MMGs

File: BDArmory/Weapons/Missiles/BDModularGuidance.cs [hunk 3]
```diff
@@ -1032,6 +1032,12 @@ public void ExecuteNextStage()
             vessel.ActionGroups.ToggleGroup(
                 (KSPActionGroup)Enum.Parse(typeof(KSPActionGroup), "Custom0" + (int)_nextStage));
 
+            if (StagesNumber == 1)
+            {
+                if (SpawnUtils.CountActiveEngines(vessel) < 1)
+                    SpawnUtils.ActivateAllEngines(vessel);
```
Comment (BrettRyland) [inline BDArmory/Weapons/Missiles/BDModularGuidance.cs:1038]: `ActivateAllEngines` will need to have the `ignoreModularMissileEngines` parameter being `false` here.

File: BDArmory/UI/BDAEditorArmorWindow.cs [hunk 4]
```diff
@@ -1039,7 +1039,7 @@ private void CalculateArmorStats()
                 */
                 //armorValue = ProjectileUtils.CalculatePenetration(30, newCaliber, 0.388f, 1109, ArmorDuctility, ArmorDensity, ArmorStrength, 30, 0.8f, false);
                 armorValue = ProjectileUtils.CalculatePenetration(30, 1109, 0.388f, 0.8f, ArmorStrength, ArmorVfactor, ArmorMu1, ArmorMu2, ArmorMu3); //why is this hardcoded? it needs to be the selected armor mat's vars
-                relValue = Mathf.Round(steelValue / armorValue * 10) / 10;
+                BDAMath.RoundToUnit(steelValue / armorValue, 0.1f)
```
Comment (BrettRyland) [inline BDArmory/UI/BDAEditorArmorWindow.cs:1042]: I'm fairly sure you didn't mean to also remove the `relValue = ` and `;` parts of this line.

**Pull request merged**